File: 768-partition-labels/partition-labels.java
class Solution {
    public List<Integer> partitionLabels(String s) {
        int LastAppeared[] = new int[26];
        List<Integer> ans = new ArrayList<>();
        
        for(int i=0;i<s.length();i++)
        {
            char currChar = s.charAt(i);
            LastAppeared[currChar - 'a'] = i;
        }
        int start = 0 ,end=0;
        for(int i=0;i<s.length();i++)
        {
            end = Math.max(end, LastAppeared[s.charAt(i)-'a']);
            if(i==end)
            {
                ans.add(i -start+1);
                start = i+1;
            }
        }
        return ans;
    }
}

File: 768-partition-labels/partition-labels.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        charcount = [0]*26
        n = len(s)
        res = []

        for i in range(n):
            charcount[ord(s[i]) - ord('a')] = i

        print(charcount)
        start =end =0
        for i in range(n):
            end = max(end, charcount[ord(s[i]) - ord('a')])
            if end==i:
                res.append(i - start+1)
                start = i +1
        
        return res

        